src/include/uapi/ipc/xtensa.h

```diff
@@ -65,6 +67,11 @@ struct sof_ipc_dsp_oops_xtensa {
 	uint32_t intenable;
 	uint32_t interrupt;
 	uint32_t sar;
+	uint32_t debugcause;
+	uint32_t widnowbase;
+	uint32_t windowstart;
+	uint32_t excsave1;
+	uint32_t ar[XCHAL_NUM_AREGS];
```

Comment:
Btw, we probably want to either add reserved[] here or add other registers (that may not be used or saved today, but will be in the future). i.e on qemu I can see
```
PC=be0103c0

        LBEG=be026645         LEND=be02664b       LCOUNT=00000000          SAR=00000019
          BR=00000000    SCOMPARE1=00000000      PREFCTL=00001044  WINDOW_BASE=0000000d
WINDOW_START=00002aaa IBREAKENABLE=00000000       MEMCTL=00000000      ATOMCTL=00000015
    IBREAKA0=00000000     IBREAKA1=00000000     DBREAKA0=00000000     DBREAKA1=00000000
    DBREAKC0=00000000     DBREAKC1=00000000    CONFIGID0=c3f3fbfe         EPC1=be01181b
        EPC2=00000000         EPC3=00000000         EPC4=00000000         EPC5=00000000
        EPC6=be0103c0         EPC7=00000000         DEPC=be010280         EPS2=00000000
        EPS3=00000000         EPS4=00000000         EPS5=00000000         EPS6=00060d35
        EPS7=00000000    CONFIGID1=1544813c     EXCSAVE1=00000000     EXCSAVE2=be026034
    EXCSAVE3=be026104     EXCSAVE4=be0261d4     EXCSAVE5=be0262a4     EXCSAVE6=00000000
    EXCSAVE7=00000000       INTSET=00000000     INTCLEAR=00000000    INTENABLE=00000098
          PS=00060d36      VECBASE=be010000     EXCCAUSE=00000000   DEBUGCAUSE=00000008
      CCOUNT=00000000         PRID=00000000       ICOUNT=00000000  ICOUNTLEVEL=00000000
    EXCVADDR=9e02e594    CCOMPARE0=00000000    CCOMPARE1=00000000    CCOMPARE2=00000000

THREADPTR=be04601c FCR=00000000 FSR=00000000 

 A00=be0118c0  A01=be07eec0  A02=0000000f  A03=be010a40
 A04=000000c0  A05=00000000  A06=00000000  A07=00000000
 A08=9e02e580  A09=9e02e580  A10=00000000  A11=00000000
 A12=e0000000  A13=be07f000  A14=fffffffd  A15=00040020

AR00=00000000 AR01=be07f000 AR02=fffffffd AR03=00040020
AR04=7e010763 AR05=be07efe0 AR06=00000000 AR07=fffffffe
AR08=e0000000 AR09=be02e448 AR10=00000000 AR11=00000000
AR12=be0112f8 AR13=be07efc0 AR14=be046058 AR15=be00e008
AR16=00000008 AR17=be00e003 AR18=00000000 AR19=00000000
AR20=be01113a AR21=be07efa0 AR22=be046058 AR23=be02e580
AR24=00000005 AR25=00000040 AR26=be02e900 AR27=00000005
AR28=be013085 AR29=be07ef80 AR30=be046058 AR31=00000001
AR32=00000000 AR33=be046044 AR34=00000000 AR35=be047508
AR36=be013e0b AR37=be07ef60 AR38=00000011 AR39=00000001
AR40=000000c0 AR41=00000098 AR42=ffffffff AR43=00060020
AR44=be01197f AR45=be07ef40 AR46=00000011 AR47=00000001
AR48=000000c0 AR49=be02e708 AR50=00060020 AR51=00000000
AR52=be0118c0 AR53=be07ef20 AR54=00000011 AR55=00000000
AR56=000000c0 AR57=00000000 AR58=00000000 AR59=00000000
AR60=00000000 AR61=00000000 AR62=00000000 AR63=00000000

F00=00000000 (+0.00000000e+00) F01=00000000 (+0.00000000e+00)
F02=00000000 (+0.00000000e+00) F03=00000000 (+0.00000000e+00)
F04=00000000 (+0.00000000e+00) F05=00000000 (+0.00000000e+00)
F06=00000000 (+0.00000000e+00) F07=00000000 (+0.00000000e+00)
F08=00000000 (+0.00000000e+00) F09=00000000 (+0.00000000e+00)
F10=00000000 (+0.00000000e+00) F11=00000000 (+0.00000000e+00)
F12=00000000 (+0.00000000e+00) F13=00000000 (+0.00000000e+00)
F14=00000000 (+0.00000000e+00) F15=00000000 (+0.00000000e+00)
```

Reply:
Oh, we also cant use XCHAL_NUM_AREGS here either, since this will mean ABI will change depending on target. Can we have a MAX num AREGS ?, We could maybe include a uint32_t called num_aegs so userspace tool knows the number of aregs.

Reply:
@lgirdwood @akloniex The core dump seems to be very arch specific. We may try to keep some pieces like the beginning of the stack content at the fixed position but would this still help the driver to provide any readable prints to the dmesg? As you mentioned, there are so many possible configurations of the core that being able to parse/scan them all inside the driver might be really challenging even if there is some common alignment applied. How about moving some common info like file/location to the beginning and use a fixed format for that, and treat the rest as cpu arch specific blob? Plugging more and more arch specific scanners (or leave it to the debug utilities), familiar with the arch specific info alignment, to the driver would be another question.

Reply:
@lgirdwood 
I've found configid defined in our sourcecode, but how should I encode the architecture? Is there some info already available to use or should I create new definitions for each platform, or introduce some new enum for this purpose?

Reply:
You could have a header struct that contained the architecture, platform and total size, stack offset (in this structure - allows ABI updates), then the body would be different for each architecture (header would always be the same).